client/core/core.go

```diff
@@ -1629,6 +1629,90 @@ func (c *Core) WalletSettings(assetID uint32) (map[string]string, error) {
 	return dbWallet.Settings, nil
 }
 
+func (c *Core) setAppPass(pw []byte) (encrypt.Crypter, error) {
+	if len(pw) == 0 {
+		return nil, fmt.Errorf("empty password not allowed")
+	}
+
+	crypter := c.newCrypter(pw)
+	err := c.db.Store(keyParamsKey, crypter.Serialize())
+	if err != nil {
+		return nil, fmt.Errorf("error storing key parameters: %w", err)
+	}
+
+	return crypter, nil
+}
+
+func (c *Core) updateWalletsEncPW(oldCrypter, newCrypter encrypt.Crypter) error {
+	c.walletMtx.Lock()
+	defer c.walletMtx.Unlock()
+	for _, w := range c.wallets {
+		w.mtx.Lock()
+		defer w.mtx.Unlock()
```

Comment:
This defer will happen at the end of the function, and not the end of the loop. I think it is fine here, but could lead to a "gotcha" if this method is ever expanded downwards.

Reply:
Thanks guys will do.
@buck54321 Nice pattern to resolve such scenarios.